src/System.Management.Automation/engine/parser/TypeInferenceVisitor.cs

```diff
@@ -581,6 +617,15 @@ object ICustomAstVisitor.VisitUnaryExpression(UnaryExpressionAst unaryExpression
         object ICustomAstVisitor.VisitConvertExpression(ConvertExpressionAst convertExpressionAst)
         {
             var type = convertExpressionAst.Type.TypeName.GetReflectionType();
+
+            if (type == typeof(PSObject) && convertExpressionAst.Child is HashtableAst hashtableAst)
```

Comment:
Do you mean to use `type == typeof(PSCustomObject)`? `[psobject]@{ a = 1 }` just wrap the hashtable into psobject, and it should still be `Hashtable` type.

Reply:
I guess `[psobject]@{a =1}` would be very rare and thus we don't need to differentiate `[pscustomeobject]@{a=1}` from `[psobject]@{a =1}`.

But can you please add a comment about `[PSCustomObject]` actually returns `PSObject` as reflection type?